# Molecular Docking Using Google Colab by Purnawan Pontana Putra

In [ ]:
!wget https://github.com/ccsb-scripps/AutoDock-Vina/releases/download/v1.2.3/vina_1.2.3_linux_x86_64

--2022-04-04 14:38:55--  https://github.com/ccsb-scripps/AutoDock-Vina/releases/download/v1.2.3/vina_1.2.3_linux_x86_64
Resolving github.com (github.com)... 140.82.113.3
Connecting to github.com (github.com)|140.82.113.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/258054635/ae531efa-011f-44eb-86d3-dc1115228edc?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20220404%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20220404T143856Z&X-Amz-Expires=300&X-Amz-Signature=e269c1790375e2f05bbb08276a5091731cd92b210376952cc6d775bd95b99caf&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=258054635&response-content-disposition=attachment%3B%20filename%3Dvina_1.2.3_linux_x86_64&response-content-type=application%2Foctet-stream [following]
--2022-04-04 14:38:56--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/258054635/ae531efa-011f-44eb-

In [ ]:
!mv vina_1.2.3_linux_x86_64 vina

In [ ]:
alias vina /content/vina

In [ ]:
!chmod 755 -R /content/vina

In [ ]:
%vina

AutoDock Vina v1.2.3
#################################################################
# If you used AutoDock Vina in your work, please cite:          #
#                                                               #
# J. Eberhardt, D. Santos-Martins, A. F. Tillack, and S. Forli  #
# AutoDock Vina 1.2.0: New Docking Methods, Expanded Force      #
# Field, and Python Bindings, J. Chem. Inf. Model. (2021)       #
# DOI 10.1021/acs.jcim.1c00203                                  #
#                                                               #
# O. Trott, A. J. Olson,                                        #
# AutoDock Vina: improving the speed and accuracy of docking    #
# with a new scoring function, efficient optimization and       #
# multithreading, J. Comp. Chem. (2010)                         #
# DOI 10.1002/jcc.21334                                         #
#                                                               #
# Please see https://github.com/ccsb-scripps/AutoDock-V

In [ ]:
%vina --receptor /content/4ieh_protein.pdbqt --ligand /content/4ieh_ligand.pdbqt --config /content/grid.txt --exhaustiveness=8 --out 4ieh_vina_out.pdbqt

AutoDock Vina v1.2.3
#################################################################
# If you used AutoDock Vina in your work, please cite:          #
#                                                               #
# J. Eberhardt, D. Santos-Martins, A. F. Tillack, and S. Forli  #
# AutoDock Vina 1.2.0: New Docking Methods, Expanded Force      #
# Field, and Python Bindings, J. Chem. Inf. Model. (2021)       #
# DOI 10.1021/acs.jcim.1c00203                                  #
#                                                               #
# O. Trott, A. J. Olson,                                        #
# AutoDock Vina: improving the speed and accuracy of docking    #
# with a new scoring function, efficient optimization and       #
# multithreading, J. Comp. Chem. (2010)                         #
# DOI 10.1002/jcc.21334                                         #
#                                                               #
# Please see https://github.com/ccsb-scripps/AutoDock-V

# Calculation RMSD 

In [ ]:
!apt install openbabel

Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libopenbabel4v5
The following NEW packages will be installed:
  libopenbabel4v5 openbabel
0 upgraded, 2 newly installed, 0 to remove and 39 not upgraded.
Need to get 2,766 kB of archives.
After this operation, 11.8 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic/universe amd64 libopenbabel4v5 amd64 2.3.2+dfsg-3build1 [2,664 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic/universe amd64 openbabel amd64 2.3.2+dfsg-3build1 [102 kB]
Fetched 2,766 kB in 1s (2,494 kB/s)
Selecting previously unselected package libopenbabel4v5.
(Reading database ... 156210 files and directories currently installed.)
Preparing to unpack .../libopenbabel4v5_2.3.2+dfsg-3build1_amd64.deb ...
Unpacking libopenbabel4v5 (2.3.2+dfsg-3build1) ...
Selecting previously unselected package openbabel.
Preparing to unpack .../openbabel

In [ ]:
!obrms -firstonly 4ieh_ligand.pdbqt 4ieh_vina_out.pdbqt

RMSD  0.783042
RMSD  2.36812
RMSD  2.29354
RMSD  2.66277
RMSD  3.57055
RMSD  3.8705
RMSD  3.14858
RMSD  12.4039
RMSD  2.56186


*The top-ranked pose is \<2A from the crystal pose. In this tutorial, I have done superpose file 4ieh_ligand.pdbqt and Mode 1 from 4ieh_vina_out.pdbqt and found the RMSD score is 0.783042. This is a good value. The RMSD consensus is <2*

# Visualization

In [ ]:
!pip install py3Dmol

In [ ]:
import py3Dmol
v = py3Dmol.view()
v.addModel(open('4ieh_protein.pdbqt').read())
v.setStyle({'cartoon':{},'stick':{'radius':.1}})
v.addModel(open('4ieh_ligand.pdbqt').read())
v.setStyle({'model':1},{'stick':{'colorscheme':'dimgrayCarbon','radius':.125}})
v.addModelsAsFrames(open('4ieh_vina_out.pdbqt').read())
v.setStyle({'model':2},{'stick':{'colorscheme':'greenCarbon'}})
v.animate({'interval':1000})
v.zoomTo({'model':1})
v.rotate(90)

You appear to be running in JupyterLab (or JavaScript failed to load for some other reason). You need to install the 3dmol extension: 
 jupyter labextension install jupyterlab_3dmol